### ISSUE TITLE
"&t=tvsearch" yielding no results in API calls (Sickrage/Sickbeard) 

### ISSUE THREAD

**Issue by nrk666** [nZEDb/nZEDb#1997]
Latest MASTER pull as of 12/10/15, DB updated, no errors. 

"&t=search" from browser works fine, yields lots of results. 

http://url/api?apikey=APIKEY$t=search&cat=5040&q=TVSHOWNAME

"&t=tvsearch" from browser yields no results. This is the API command that Sickrage/Sickbeard uses to search for shows, so nothing is getting pulled. 

http://url/api?apikey=APIKEY&t=tvsearch&cat=5040&q=TVSHOWNAME

You get nothing.

**Comment by nrk666** [comment]
Did a completely fresh install from scratch, same problem. &t=tvsearch returns no results.

**Comment by DariusIII** [comment]
I just tested and it works just fine. Enable debugging in settings.php, enable all the debugging options.

**Comment by ruhllatio** [comment]
What is your Search type set to?  Are you standard, MySQL FT, or Sphinx?

**Comment by Fossil01** [comment]
The api template files changed. Might be the issue.

**Comment by nrk666** [comment]
Search Type is set to "1" for standard. 

I set the following fields in settings.php to true: 

nZEDb_LOGGING -> true
nZEDb_LOGINFO -> true
nZEDb_LOGNOTICE -> true
nZEDb_LOGWARNING -> true
nZEDb_LOGERROR -> true
nZEDb_LOGFATAL -> true
nZEDb_LOGQUERIES -> true

I then ran a &t=tvsearch query, got zero results, but no debug log was created. 

The &t=search API call retrurned many results, but Sickrage used &t=tvsearch.

**Comment by ruhllatio** [comment]
If the query is passing, there won't be a debug log for it.  To be sure make sure you're looking for php_errors and nzedb.log in nZEDb_ROOT/resources/logs.  You need to var_dump() the query it is spitting out.  FWIW, we have too many configurations to test and it might not be returning for MySQL FT though I didn't change that.  Most of us testing it and saying it works fine are using Sphinx as it is far, far superior.

**Comment by nrk666** [comment]
This is just a small private instance for myself, Sphinx is overkill. I only index like 6 groups, primarily for Sickrage. It was also working prior to this last big update. Sorry, "1" isn't standard, I am assuming "1" is for MYSQL FT. 

nzedb.log is only showing me NNTP connections, Binary update status, etc, I don't see any error messages in there. 

I did "find /var/www/nZEDb -name php_errors.log" and examined all of them, none of the entries are timestamped even close to when I just ran this failed query. 

I will try to switch to Sphinx to try isolate if that helps or not, but I'd really appreciate it if you could look into fixing MYSQL FT.

**Comment by Fossil01** [comment]
Sphinx can be running in like 15 mins. I would just do it.

**Comment by niel** [comment]
@Fossil true, for those that know what they're doing and don't run into any problems. ;-)

**Comment by Wally73** [comment]
howto install sphinx: 

https://github.com/nZEDb/nZEDb/tree/master/misc/sphinxsearch

**Comment by nrk666** [comment]
Ok, I've installed/configured Sphinxsearch, it appears to be working fine from the WEB UI. When I "search" through the WebUI it comes back with a ton of "Sphinxsearch Options" for Advandced Searching. 

To test this, I took a TV show, "Gotham" for example, and I moved one of the Sickrage shows to a new dir and set the Episode in Sickrage to "Wanted" so that it would force a search for that particular show. 

From the Sickrage Debug log it ran these API queries: 

http://NZEDBSERVER/api?apikey=**********&maxage=4&cat=5030%2C5040&limit=100&t=tvsearch&offset=0

Which returned nothing. 

Sickrage then ran this much more specific API query: 

http://NZEDBSERVER/api?apikey=**********&tvdbid=274431&season=2&maxage=12&cat=5030%2C5040&q=Gotham&limit=100&t=tvsearch&offset=0&ep=11

Also returns nothing. 

If I manually change "&t=tvsearch" to "&t=search" then it returns the values I'd expect. 

This is with Sphinxsearch ON (settings.php updated to use "2"). I seem to be getting the same failed queries with both MySQL FT and Sphinxsearch. Also, this is on a 100% fresh install of both the DB and nZEDb MASTER branch as of last night. The only thing I did was import all my old NZB from my old tree, so "Gotham" is actually in there,.

**Comment by Bart39** [comment]
Have you actually post processed to yet ? That would explain why no results

**Comment by nrk666** [comment]
Yes, post processing ran all night, and like I said, if I just change "&t=tvsearch" to "&t=search" doing a manual browser API query, I get all kinds of results. But, Sickrage uses "&t=tvsearch", I have no control over that. 

If I use the same "&t=tvsearch" API URL to a private external indexer, I get results (changing url and APIKEY of course).

**Comment by Bart39** [comment]
`SELECT COUNT(*) FROM releases WHERE nzbstatus = 1 AND categoryid BETWEEN 5000 AND 5999 AND categoryid != 5070 AND videos_id = 0 AND tv_episodes_id BETWEEN -3 AND 0 AND size > 1048576`

**Comment by nrk666** [comment]
MariaDB [nzedb]> SELECT COUNT(*) FROM releases WHERE nzbstatus = 1 AND categoryid BETWEEN 5000 AND 5999 AND categoryid != 5070 AND videos_id = 0 AND tv_episodes_id BETWEEN -3 AND 0 AND size > 1048576;
+----------+
| COUNT(*) |
+----------+
|   239212 |
+----------+
1 row in set (0.00 sec)

MariaDB [nzedb]>

**Comment by nrk666** [comment]
I have no idea what that query means, so if that was to illustrate some point, you'll have to be more specific. Is that number good or bad?

**Comment by ruhllatio** [comment]
You didn't enable TV post processing.  That's what @Bart39 meant.  There are two issues here.

1) What SickRage is doing in the first query is basically an RSS feed type pull.  It's looking for a list of recent releases to parse so that it can fulfil multiple searches with one API hit.  This SHOULD return results without TV post processing and I have identified the issue there.  Will test and push to Dev branch as this is how it was supposed to work as a fallback.  One could argue that they should be using the actual RSS functions for this, but I digress... all  of the scrapers do it this way AFAIK.

2) The second query is a directed search.  Unfortunately for SickRage, they are querying both the tvdbid/season/episode as well as the title of the show.  This may produce undesirable results as all parameters have to match if specified with the exception of multiple scraped Site IDs (If I submit the query with the TVDB ID AND the TVMaze ID for Gotham, the query will return matches for both IDs).  This will NEVER work if you don't enable TV processing.  Backlog searches are all done this way, so if you ever intend to do that, TV processing must be enabled.  There is no other way to provide the Season/Episode data in the new TV processing in nZEDb.

So in short, go into Site->Edit and make sure TV processing is on.  If you're running Tmux, post-process non-Amazon needs to be enabled as well.

**Comment by nrk666** [comment]
Ok, thanks, that was much more helpful. I'll turn that on. 

That might actually be my issue. I remember when TVRage went down I turned that off because it just kept failing anyway, and since this is a new install, it probably didn't have the local cache built up like before. 

I'll let this thing plug away for a couple days (I seem to remember this takes a while) and see whats up, but that does a ring a bell, you are probably right.

**Comment by ruhllatio** [comment]
TVRage is gone and we didn't use any of the old stuff.  So yeah, you have no local cache.

**Comment by Bart39** [comment]
The reason I wanted you to run the query was to prove what I said in the first place - you hadn't post processed any tv hence why tvsearch didn't return anything.
Just got side tracked with my day job so couldn't reply earlier
